### DISCUSSION TITLE
Expose loader breaks with dynamic imports

### DISCUSSION THREAD

**Discussion by dgarciabriseno** [webpack/webpack#17796] [Q&A]
This might be a bug, but it might also be user error, so I'm hoping to get some help.

I'm running into this issue:
1. ExposedClass is exposed via expose loader with webpack.
2. OtherClass is imported dynamically.
3. ExposedClass is gone from global scope.

Has anyone had this issue? Here's a [demo of the problem](https://github.com/dgarciabriseno/dynamic-import-expose-loader)

**Comment by alexander-akait** [comment]
Sorry for delay, but you have `override: true`, so `expose-loader` override global value, I think you forgot to use `test` for `expost-loader` to avoid this, i.e.:
```js
module: {
    rules: [
        {
            test: /exposed_class\.js$/,
            use: [
                {
                    loader: "expose-loader",
                    options: {
                      exposes: [
                        {
                            globalName: "ExposedClass",
                            moduleLocalName: "ExposedClass",
                            override: true
                        },
                      ]
                    }
                },
            ]
        }
      ]
  }
```
